charts/Util.ts

```diff
@@ -423,3 +423,30 @@ export function urlToSlug(url: string): string {
 export function sign(n: number) {
     return n > 0 ? 1 : n < 0 ? -1 : 0
 }
+
+export async function fetchJSON(url: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+        const req = new XMLHttpRequest()
+        req.addEventListener("load", function() {
+            resolve(JSON.parse(this.responseText))
+        })
+        req.addEventListener("readystatechange", () => {
+            if (req.readyState === 4) {
+                if (req.status !== 200) {
+                    reject(new Error(`${req.status} ${req.statusText}`))
+                }
+            }
+        })
+        req.open("GET", url)
+        req.send()
+    })
+    // window.fetch() implementation below. Decided to use XMLHttpRequest for
+    // the time being, since mocking window.fetch() seemed not to allow
+    // specifying an endpoint, rather you just intercept function calls, which
+    // feels more verbose.
```

Comment:
Yeah, this is weird, the Jest mock stuff and even the packages explicitly for mocking fetch don't seem to work the way xhr-mock works. I agree with you that specifying url patterns is better. We could easily roll our own stuff for this using Jest `mock`, and I think using `fetch` would be more elegant than XHRs, but this is fine for now